mock/service/controller.go

```diff
@@ -326,6 +326,38 @@ func (s *service) ValidateVolumeCapabilities(
 	}, nil
 }
 
+func (s *service) ControllerGetVolume(
+	ctx context.Context,
+	req *csi.ControllerGetVolumeRequest) (
+	*csi.ControllerGetVolumeResponse, error) {
+
+	if hookVal, hookMsg := s.execHook("GetVolumeStart"); hookVal != codes.OK {
+		return nil, status.Errorf(hookVal, hookMsg)
+	}
+
+	i, v := s.findVolByID(ctx, req.VolumeId)
+	if i < 0 {
+		return nil, status.Error(codes.NotFound, req.VolumeId)
+	}
+
+	resp := &csi.ControllerGetVolumeResponse{
+		Volume: &v,
+		Status: &csi.ControllerGetVolumeResponse_VolumeStatus{},
+	}
+
+	volumeStatus, err := s.NodeGetVolumeStats(ctx, &csi.NodeGetVolumeStatsRequest{})
```

Comment:
Why is NodeGetVolumeStats called within ControllerGetVolume?

Reply:
I misunderstand the working mechanism between Controller and Agent. Revised